feat: Integration tests workflow

MichaelsJP opened the pull request:
### Pull Request Checklist
<!--- Please make sure you have completed the following items BEFORE submitting a pull request (put an x in each box
when you have checked you have done them): -->
- [ ] 1. I have [**rebased**][rebase] the latest version of the main branch into my feature branch and all conflicts
         have been resolved.
- [ ] 2. I have added information about the change/addition to functionality to the CHANGELOG.md file under the
         [Unreleased] heading.
- [ ] 3. I have documented my code using JDocs tags.
- [ ] 4. I have removed unnecessary commented out code, imports and System.out.println statements.
- [ ] 5. I have written JUnit tests for any new methods/classes and ensured that they pass.
- [ ] 6. I have created API tests for any new functionality exposed to the API.
- [ ] 7. If changes/additions are made to the ors-config.json file, I have added these to the [ors config documentation][config]
         along with a short description of what it is for, and documented this in the Pull Request (below).
- [ ] 8. I have built graphs with my code of the Heidelberg.osm.gz file and run the api-tests with all test passing
- [ ] 9. I have referenced the Issue Number in the Pull Request (if the changes were from an issue).
- [ ] 10. For new features or changes involving building of graphs, I have tested on a larger dataset
          (at least Germany), and the graphs build without problems (i.e. no out-of-memory errors).
- [ ] 11. For new features or changes involving the graphbuilding process (i.e. changing encoders, updating the
          importer etc.), I have generated longer distance routes for the affected profiles with different options
          (avoid features, max weight etc.) and compared these with the routes of the same parameters and start/end
          points generated from the current live ORS.
          If there are differences then the reasoning for these **MUST** be documented in the pull request.
- [ ] 12. I have written in the Pull Request information about the changes made including their intended usage
          and why the change was needed.
- [ ] 13. For changes touching the API documentation, I have tested that the API playground [renders correctly][api].

Fixes # .

### Information about the changes
- Key functionality added:
- Reason for change:

### Examples and reasons for differences between live ORS routes, and those generated from this pull request
-

### Required changes to ors config (if applicable)
-

[config]: https://GIScience.github.io/openrouteservice/run-instance/configuration/
[api]: https://gitlab.gistools.geog.uni-heidelberg.de/giscience/openrouteservice-infrastructure/ors-docs-api#test-new-ors-documentation
[rebase]: https://github.com/GIScience/openrouteservice/blob/main/CONTRIBUTE.md#pull-request-guidelines

koebi commented on docker-entrypoint.sh:
```diff
@@ -183,7 +183,7 @@ fi
 ors_engine_elevation_cache_path=$(env | grep "^ors\.engine\.elevation\.cache_path=" | awk -F '=' '{print $2}')
-# get ors.engine.source_file
+# get ors.engine.profile_default.build.source_file
 ors_engine_source_file=$(env | grep "^ors\.engine\.source_file=" | awk -F '=' '{print $2}')
```
if the comment changes, shouldn't the code change as well?

MichaelsJP replied:
Yep. That was some halfway through fixing. The PR is not ready yet. I will do the docker fix now.

MichaelsJP committed:
feat(integration-tests): Isolate the integration tests

This commit truly isolates the integration tests scenarios. Before testcontainers would always randomly cache unrelated parts of the dockerfiles. With this setup it doesn't happen anymore and the CI workflow will be able to run way faster.

MichaelsJP committed:
CI(integration-tests): Add the integration-tests.yml workflow

The new workflow first builds and caches the builder image. This image is big and clunky and would only increase the parallel test time massively. The testing graph is build in the same way. The image ID from the image is used to check compatibility with the graph. When all that works the integration tests are started in parallel for all scenarios and test classes.

MichaelsJP committed:
CI(integration-tests): Only run on main

The tests are too heavy to be run all the time. Let's limit them to PR and push on main

MichaelsJP committed:
docs(integration-tests): Introduce docs for integration tests

The integration tests are quite powerful but also more complex than usual tests. Hence, they need some deeper explanation.
